packages/browser/__tests__/login/oidc/TokenRequester.spec.ts

```diff
@@ -46,6 +46,16 @@ const mockJWK = {
   d: "yR1bCsR7m4hjFCvWo8Jw3OfNR4aiYDAFbBD9nkudJKM",
 };
 
+// Allow our tests to control the JWT that comes back from our mock.
+// Setting a file scoped variable like this relies on our test runner running
+// tests within a single file in series, and not in parallel!
+enum DecodeJwtResult {
+  Undefined,
+  SubClaimPresent,
+  Empty,
+}
+let decodeJwt: DecodeJwtResult = DecodeJwtResult.SubClaimPresent;
```

Comment:
Certainly sounds very good - but given the comment on line 317 (next comment) - i.e., the entire file is due to be removed, I assume this isn't worth doing here now at all, right?! I would be very interested in applying this 'better' behaviour elsewhere though (and therefore learning how it actually works!) - so do you know anywhere applicable in the codebase right now (so that we can at least copy-and-paste your excellent explanation and comments there ('cos I can't be bothered to read and understand them right now, since I don't have a strong motivation to do so (since this file is due to be deleted) :) !)...?

Reply:
Maybe it could be used in the tests for `AuthCodeRedirectHandler`. I think I would remove the current file and `TokenRefresher`, the file that depends on this one, altogether, rather than fixing the code coverage here.